Emby.Naming/Video/DateTimeResolvers/SelfShotMoviesMovieDateTimeResolver.cs

```diff
@@ -0,0 +1,53 @@
+using System.Linq;
+using System.Text.RegularExpressions;
+using Emby.Naming.Common;
+
+namespace Emby.Naming.Video.DateTimeResolvers;
+
+/// <summary>
+/// Self-Shot videos often have a timestamp in them. Here we want to preserve the name.
+/// Eg. "My Movie 2013.12.09".
+/// If there is a additional timestamp we will use the extra date but not.
+/// </summary>
+public partial class SelfShotMoviesMovieDateTimeResolver : IMovieDateTimeResolver
+{
+    [GeneratedRegex(@"(?'name'.*(?'timestamp'(19[0-9]{2}|20[0-9]{2})(-|\.)?([0-9]{1,2})(-|\.)?([0-9]{1,2}))).*(?'date'19[0-9]{2}|20[0-9]{2})", RegexOptions.IgnoreCase)]
+    private static partial Regex TimestampWithDateRegex();
+
+    [GeneratedRegex(@"(?'name'.*(?'timestamp'(19[0-9]{2}|20[0-9]{2})(-|\.)?([0-9]{1,2})(-|\.)?([0-9]{1,2})))", RegexOptions.IgnoreCase)]
+    private static partial Regex TimestampRegex();
+
+    /// <summary>
+    /// Attempts to resolve date and Name from the provided fileName.
+    /// </summary>
+    /// <param name="fileName">Name of video.</param>
+    /// <param name="namingOptions">NamingOptions.</param>
+    /// <returns>null if could not resolve.</returns>
+    public CleanDateTimeResult? Resolve(string fileName, NamingOptions namingOptions)
+    {
+        foreach (var regex in new[] { TimestampWithDateRegex(), TimestampRegex() })
```

Comment:
Move this array to `private static readonly`

Reply:
extracted field in latest commit